src/cpp/gc-data.h

```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#include "spin-mutex.h"
+#include "gc-object.h"
+
+#include <unordered_set>
+
+namespace effil {
+
+// Base class for data represented in Lua.
+// Derived classes always managed by corresponding views.
+class GCData {
+public:
+    GCData() = default;
+    virtual ~GCData() = default;
+
+    // List of weak references to nested objects
+    std::unordered_set<GCHandle> refers() const;
+
+public://protected:
+    void addReference(GCHandle handle);
+    void removeReference(GCHandle handle);
+
+public:
+    GCData(const GCData&) = delete;
+    GCData& operator=(const GCData&) = delete;
+
+private:
+    mutable SpinMutex mutex_;
+    std::unordered_multiset<GCHandle> weakRefs_;
```

Comment:
why it's multiset?

Reply:
Multiset haps us store several week refs to the same object, thus following code works:
```
test.gc.store_same_value = function()
    local fill = function (c)
        local a = effil.table {}
        c:push(a)
        c:push(a)
    end

    local c = effil.channel()
    fill(c)

    c:pop()
    collectgarbage()
    effil.gc.collect()
    c:pop()[1] = 0
end
```